### PR TITLE
add selection fo matrix

### PR DESCRIPTION
**Pull request by AndrewLinkii**

### CODE REVIEW

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 1]
```diff
@@ -6,14 +6,44 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] mainDiagonl = new int[matrix.length];
+
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]: remove redundant laine

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 2]
```diff
@@ -6,14 +6,44 @@
...
+        for (int i = 0; i < matrix.length; i++) {
+            for (int j = 0; j < matrix[i].length; j++) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]: do we need two for loops?

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 3]
```diff
@@ -6,14 +6,44 @@
...
+                if (i == j) {
+                    mainDiagonl[i] = matrix[i][j];
+                }
+            }
+        }
+
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]: remove redundant line

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 4]
```diff
@@ -6,14 +6,44 @@
...
+        return mainDiagonl;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int tempMin = 0;
+
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]: line)

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 5]
```diff
@@ -6,14 +6,44 @@
...
+                }
+            }
+        }
+
+        return mainDiagonl;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int tempMin = 0;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]:
```suggestion
        int min = 0;
```
why min = 0?

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 6]
```diff
@@ -6,14 +6,44 @@
...
+                }
+            }
+        }
+
+        return tempMin;
     }
 
     public int getMax(int[][] matrix) {
-        return 0;
+        int tempMax = -5;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]: why tempMax = 5?
```suggestion
        int max = -5;
```

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 7]
```diff
@@ -6,14 +6,44 @@
...
+        int tempMin = 0;
+
+        for (int i = 0; i < matrix.length; i++) {
+            for (int j = 0; j < matrix[i].length; j++) {
+                if (matrix[i][j] < tempMin) {
+                    tempMin = matrix[i][j];
+                }
+            }
+        }
+
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]: remove line)

**Comment by AndrewLinkii** [comment]
fixed

**Pull request closed without merging**